src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,205 @@ type Props = {
   event: Event;
   project: Project;
   location: Location;
-
+  api: Client;
   group?: Group;
   className?: string;
 } & typeof defaultProps;
 
-class EventEntries extends React.Component<Props> {
+type State = {
+  isLoading: boolean;
+  proGuardErrors: ProGuardErrors;
+};
+
+class EventEntries extends React.Component<Props, State> {
   static defaultProps = defaultProps;
 
+  state: State = {
+    isLoading: true,
+    proGuardErrors: [],
+  };
+
   componentDidMount() {
+    this.checkProGuardError();
+    this.recordIssueError();
+  }
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    const {event, showExampleCommit} = this.props;
+
+    return (
+      (event && nextProps.event && event.id !== nextProps.event.id) ||
+      showExampleCommit !== nextProps.showExampleCommit ||
+      nextState.isLoading !== this.state.isLoading
+    );
+  }
+
+  async fetchProguardMappingFiles(query: string): Promise<Array<DebugFile>> {
+    const {api, organization, project} = this.props;
+    try {
+      const proguardMappingFiles = await api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/files/dsyms/`,
```

Comment:
I think this might get rejected on a shared event view - should we put this behind a condition?

Reply:
but I tested it locally and it was fine

Reply:
yes..Mark suggested the same. I will update the code 😉